src/main/java/io/cdap/plugin/http/source/common/BaseHttpSourceConfig.java

```diff
@@ -0,0 +1,931 @@
+          }
+          break;
+        case CUSTOM:
+          propertiesShouldBeNotNull.put(PROPERTY_CUSTOM_PAGINATION_CODE,
+                                        propertiesShouldBeNull.remove(PROPERTY_CUSTOM_PAGINATION_CODE));
+          break;
+        // other types don't require any fields. Check for unknown values is already done. Do nothing here
+      }
+
+      for (Map.Entry<String, Object> entry : propertiesShouldBeNull.entrySet()) {
+        String propertyName = entry.getKey();
+        Object propertyValue = entry.getValue();
+        assertIsNotSet(propertyValue, propertyName, reasonPagination);
+      }
+
+      for (Map.Entry<String, Object> entry : propertiesShouldBeNotNull.entrySet()) {
+        String propertyName = entry.getKey();
+        Object propertyValue = entry.getValue();
+        assertIsSet(propertyValue, propertyName, reasonPagination);
+      }
+    }
+
+
+    // Validate format properties
+    if (!containsMacro(PROPERTY_FORMAT)) {
+      String reasonFormat = String.format("page format is '%s'", getFormat());
+
+      if (getFormat().equals(PageFormat.JSON) || getFormat().equals(PageFormat.XML)) {
+        assertIsSet(getResultPath(), PROPERTY_RESULT_PATH, reasonFormat);
+        getFullFieldsMapping(); // can be null, but call getter to verify correctness of regexps
+      } else {
+        assertIsNotSet(getResultPath(), PROPERTY_RESULT_PATH, reasonFormat);
+        assertIsNotSet(getFieldsMapping(), PROPERTY_FIELDS_MAPPING, reasonFormat);
+      }
+    }
+
+    // Validate OAuth2 properties
+    if (!containsMacro(PROPERTY_OAUTH2_ENABLED) && this.getOauth2Enabled()) {
+      String reasonOauth2 = "OAuth2 is enabled";
+      assertIsSet(getAuthUrl(), PROPERTY_AUTH_URL, reasonOauth2);
+      assertIsSet(getTokenUrl(), PROPERTY_TOKEN_URL, reasonOauth2);
+      assertIsSet(getClientId(), PROPERTY_CLIENT_ID, reasonOauth2);
+      assertIsSet(getClientSecret(), PROPERTY_CLIENT_SECRET, reasonOauth2);
+      assertIsSet(getRefreshToken(), PROPERTY_REFRESH_TOKEN, reasonOauth2);
+    }
+
+    if (!containsMacro(PROPERTY_VERIFY_HTTPS) && !getVerifyHttps()) {
+      assertIsNotSet(getTrustStoreFile(), PROPERTY_TRUSTSTORE_FILE,
+                     String.format("trustore settings are ignored due to disabled %s", PROPERTY_VERIFY_HTTPS));
+    }
+  }
+
+  public void validateSchema() {
+    if (!containsMacro(PROPERTY_FORMAT)) {
+      try {
+        StringToRecordConverterFactory.createInstance(this, getSchema()).validateSchema();
+      } catch (UnsupportedTypeException e) {
+        // Copy message so that UI shows it instead of some other message.
+        throw new InvalidStageException(e.getMessage(), e);
+      }
+    }
+  }
+
+  protected static void assertIsSet(Object propertyValue, String propertyName, String reason) {
+    if (propertyValue == null) {
+      throw new InvalidConfigPropertyException(
+        String.format("Property %s must be set, since %s", propertyName, reason), propertyName);
```

Comment:
%s -> '%s'

Reply:
Done.

Ps. Changed to
`Property '%s' must be set, since %s`
Second value is a phrase.